### ISSUE TITLE
Python3.12(on Ubuntu 24.04) SyntaxWarning issues

### ISSUE THREAD

**Issue by birlorg** [piku/piku#346]
I'm brand new to piku, and setting it up on on Ubuntu 24.04 LTS, which runs python 3.12 now(Python 3.12.3).

In Python3.12 deprecated escape sequence issues have been upgraded to SyntaxWarning
```
/home/piku/piku.py:110: SyntaxWarning: invalid escape sequence '\.'
  NGINX_COMMON_FRAGMENT = """
/home/piku/piku.py:216: SyntaxWarning: invalid escape sequence '\*'
  CRON_REGEXP = "^((?:(?:\*\/)?\d+)|\*) ((?:(?:\*\/)?\d+)|\*) ((?:(?:\*\/)?\d+)|\*) ((?:(?:\*\/)?\d+)|\*) ((?:(?:\*\/)?\d+)|\*) (.*)$"
/home/piku/piku.py:852: SyntaxWarning: invalid escape sequence '\.'
  env['PIKU_INTERNAL_NGINX_BLOCK_GIT'] = "" if env.get('NGINX_ALLOW_GIT_FOLDERS') else "location ~ /\.git { deny all; }"
```
It's been deprecated since Python 3.6:

> A backslash-character pair that is not a valid escape sequence now generates a [DeprecationWarning](https://docs.python.org/dev/library/exceptions.html#DeprecationWarning). Although this will eventually become a [SyntaxError](https://docs.python.org/dev/library/exceptions.html#SyntaxError), that will not be for several Python releases. (Contributed by Emanuel Barry in [bpo-27364](https://bugs.python.org/issue?@action=redirect&bpo=27364).)

- https://docs.python.org/dev/whatsnew/3.6.html#deprecated-python-behavior

**Comment by rcarmo** [comment]
Thanks. Those should be just a matter of adding the 'r' prefix, but I will test them on a 24.04 machine.

**Comment by birlorg** [comment]
I agree, it seems to work for me with `r` prefix, but I don't really know how piku is supposed to work, so maybe I broke stuff I don't even know about.

**Comment by rcarmo** [comment]
It’s not broken-the ‘r’ prefix denotes regular expression rules apply to those string blocks.

**Comment by rcarmo** [comment]
Closed via #283.

**Comment by abdulbasetbasher** [comment]
```
ssh piku@xxxxxx

/home/piku/piku.py:977: SyntaxWarning: invalid escape sequence '\.'
  nginx_config_test = str(check_output("nginx -t 2>&1 | grep -E '{}\.conf:[0-9]+$'".format(app), env=environ, shell=True))
```

i think this related to same bug on Python3.12
